pkg/cmd/server/api/types.go

```diff
@@ -47,10 +47,14 @@ type MasterConfig struct {
 	// CORSAllowedOrigins
 	CORSAllowedOrigins []string
 
+	// MasterPublicURL is how clients can access the OpenShift API server
+	MasterPublicURL string
```

Comment:
Add validation to ensure the URL is present (required for swagger), well-formed, and matches any other MasterPublicURLs that may be present.